app/models/member.rb

```diff
@@ -121,6 +121,26 @@ def user_has_namespace_maintainer_access?(user, namespace)
                        access_level: Member::AccessLevel::MAINTAINER)
       end
     end
+
+    def membership_source(namespace, user)
+      if Member.exists?(
+        namespace:, user:
+      )
+        { label: I18n.t('activerecord.models.member.direct') }
+      else
+        source_namespaces = if namespace.project_namespace?
+                              namespace.parent&.self_and_ancestor_ids
+                            else
+                              namespace.self_and_ancestor_ids
+                            end
+
+        source_namespace = Namespace.find_by(id: Member.where(namespace: source_namespaces,
+                                                              user:).order(:access_level).last.namespace_id)
+
+        { inherited_namespace_path: Rails.application.routes.url_helpers.group_path(source_namespace, only_path: true),
```

Comment:
we should be using the `group_url` instead of `group_path` as this would break deployments under a subpath.

Reply:
Updated in 0379505201abd0d8b0a317f46c1819e1c82f5c3e